sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/json/JsonExpressionUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.json;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+import org.apache.spark.sql.catalyst.expressions.SharedFactory;
+import org.apache.spark.sql.catalyst.json.CreateJacksonParser;
+import org.apache.spark.unsafe.types.UTF8String;
+
+public class JsonExpressionUtils {
+
+  public static Integer lengthOfJsonArray(UTF8String json) {
+    // return null for null input
+    if (json == null) {
+      return null;
+    }
+    try (JsonParser jsonParser =
```

Comment:
The old implementation calls the `close()` method of `jsonParser` inside of  `Utils.tryWithResource` but why don't you call it.

Reply:
- Because this is Java code, the code pattern used is:
```java
try (create_resource) {
} catch(...) {
}
```

- Based on my understanding, the resources created in 'try(...)' will automatically call `resource#close()`
ref: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

- Example:
```java
public class JavaTryResourceCloseTest {

  public static void main(String[] args) {
    try (JsonParser jsonParser = new JsonParser()) {
      System.out.println("Execute try inner");
    } catch (IOException e) {

    }
    System.out.println("end");
  }
}

class JsonParser implements Closeable {

  @Override
  public void close() throws IOException {
    System.out.println("close JsonParser");
  }
}
```

- The above output is:
```shell
Execute try inner
close JsonParser
end
```

Reply:
I see, thank you.

Reply:
Thank you very much for your review!
After this PR, I want to refactor the expression `JsonObjectKeys` in the same way, as it can also support `codegen` through this approach.